torch/signal/windows/windows.py

```diff
@@ -0,0 +1,315 @@
+import torch
+
+from torch import Tensor
+from torch._torch_docs import factory_common_args
+from torch.types import _dtype, _device, _layout
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+
+def _add_docstr(*args):
+    r"""Adds docstrings to a given decorated function.
+
+    Specially useful when then docstrings needs string interpolation, e.g., with
+    str.format().
+    REMARK: Do not use this function if the docstring doesn't need string
+    interpolation, just write a conventional docstring.
+
+    Args:
+        args (str):
+    """
+    def decorator(o):
+        o.__doc__ = ""
+        for arg in args:
+            o.__doc__ += arg
+        return o
+    return decorator
+
+
+def _window_function_checks(function_name: str, window_length: int, dtype: _dtype, layout: _layout) -> None:
+    r"""Performs common checks for all the defined windows.
+     This function should be called before computing any window
+
+     Args:
+         function_name (str): name of the window function.
+         window_length (int): length of the window.
+         dtype (:class:`torch.dtype`): the desired data type of the window tensor.
+         layout (:class:`torch.layout`): the desired layout of the window tensor.
+     """
+
+    def is_floating_type(t: _dtype) -> bool:
+        return t == torch.float32 or t == torch.bfloat16 or t == torch.float64 or t == torch.float16
+
+    def is_complex_type(t: _dtype) -> bool:
+        return t == torch.complex64 or t == torch.complex128 or t == torch.complex32
+
+    if window_length < 0:
+        raise ValueError(f'{function_name} requires non-negative window_length, got window_length={window_length}')
+    if layout is not torch.strided:
+        raise ValueError(f'{function_name} is not implemented for sparse types, got: {layout}')
+    if not is_floating_type(dtype) and not is_complex_type(dtype):
+        raise ValueError(f'{function_name} expects floating point dtypes, got: {dtype}')
```

Comment:
Expects a floating point or complex dtype?

Reply:
I got these checks from C++ implementations. However, it doesn't. make sense that we support complex `dtypes`. For instance, `torch.kaiser_window` doesn't work with `torch.complex128`.

```python
>>> torch.kaiser_window(10, dtype=torch.complex128)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "arange_cpu" not implemented for 'ComplexDouble'
```

Reply:
That is very odd! Maybe a small cleanup opportunity

Reply:
Yes, sir. I think we'll end up migrating the kaiser window to Python so it'll be automatically cleaned up.